x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_details/monitor_selector/monitor_selector.tsx

```diff
@@ -0,0 +1,152 @@
+
+  const { values, loading } = useMonitorName({ search: searchValue });
+
+  const recentlyViewed = useRecentlyViewedMonitors();
+
+  useEffect(() => {
+    let newOptions: EuiSelectableOption[] = [];
+    if (recentlyViewed.length > 0 && !searchValue) {
+      const otherMonitors = values.filter((value) =>
+        recentlyViewed.every((recent) => recent.key !== value.key)
+      );
+
+      if (otherMonitors.length > 0) {
+        newOptions.push({ key: 'monitors', label: OTHER_MONITORS, isGroupLabel: true });
+      }
+
+      newOptions = [...newOptions, ...otherMonitors];
+
+      setOptions([...recentlyViewed, ...newOptions]);
+    } else {
+      setOptions(values);
+    }
+  }, [recentlyViewed, searchValue, values]);
+
+  const onButtonClick = () => {
+    setIsPopoverOpen(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setIsPopoverOpen(false);
+  };
+
+  const button = (
+    <EuiButtonIcon iconType="arrowDown" onClick={onButtonClick} aria-label={SELECT_MONITOR} />
+  );
+
+  return (
+    <Fragment>
+      <EuiPopover
+        id={'monitorSelector'}
+        panelPaddingSize="none"
+        button={button}
+        isOpen={isPopoverOpen}
+        closePopover={closePopover}
+      >
+        <EuiPopoverTitle paddingSize="s">{GO_TO_MONITOR}</EuiPopoverTitle>
+        <EuiSelectable
+          searchable
+          isLoading={loading}
+          searchProps={{
+            placeholder: PLACEHOLDER,
+            compressed: true,
+            onChange: (val) => setSearchValue(val),
+          }}
+          options={options}
+          onChange={(selectedOptions) => {
+            setOptions(selectedOptions);
+            closePopover();
+          }}
+          singleSelection={true}
+          listProps={{
+            showIcons: false,
+          }}
+          renderOption={(option, search) => (
+            <EuiLink href={`${basePath}/app/synthetics/monitor/${option.key}`}>
+              <EuiHighlight search={searchValue}>{option.label}</EuiHighlight>
+            </EuiLink>
+          )}
+          noMatchesMessage={NO_RESULT_FOUND}
+          emptyMessage={
+            <EuiButtonEmpty href={`${basePath}/app/synthetics/add-monitor`}>
+              {CREATE_NEW_MONITOR}
+            </EuiButtonEmpty>
+          }
+        >
+          {(list, search) => (
+            <div style={{ width: 280 }}>
+              <EuiPopoverTitle paddingSize="s">
+                {options.length > 0 || searchValue ? (
+                  search
+                ) : (
+                  <EuiText color="subdued" size="s" className="eui-textCenter">
+                    {NO_OTHER_MONITORS_EXISTS}
+                  </EuiText>
+                )}
+              </EuiPopoverTitle>
+              {list}
+            </div>
+          )}
+        </EuiSelectable>
+      </EuiPopover>
+    </Fragment>
+  );
+};
+
+const GO_TO_MONITOR = i18n.translate('xpack.synthetics.monitorSummary.goToMonitor', {
+  defaultMessage: 'Go to monitor',
```

Comment:
The label is "Create Monitor" in the design. However I think "Create monitor" (with small 'm') would be better.